src/fermilib/utils/_jellium_trotter_error.py

```diff
@@ -0,0 +1,461 @@
+                    double_com = double_commutator(
+                        terms[alpha], terms[beta], terms[alpha_prime])
+
+                    if alpha == beta:
+                        double_com /= 2.0
+
+                    error_operator += double_com
+
+    error_operator /= 12.0
+    return error_operator
+
+
+def jellium_error_bound(terms, indices=None, is_hopping_operator=None,
+                        tight=False):
+    """
+    Numerically upper bound the error in the ground state energy
+    for the second order Trotter-Suzuki expansion.
+
+    Args:
+        terms: a list of single-term FermionOperators in the Hamiltonian
+            to be simulated.
+        indices: a set of indices the terms act on in the same order as terms.
+        is_hopping_operator: a list of whether each term is a hopping operator.
+        tight: whether to use the triangle inequality to give a loose
+            upper bound on the error (default) or to calculate the
+            norm of the error operator.
+
+    Returns:
+        A float upper bound on norm of error in the ground state energy.
+
+    Notes: follows Poulin et al.'s work in "The Trotter Step Size
+           Required for Accurate Quantum Simulation of Quantum
+           Chemistry". In particular, Equation 16 is used for a loose
+           upper bound, and the norm of Equation 9 is calculated for
+           a tighter bound using the error operator from
+           jellium_error_operator.
+
+           Possible extensions of this function would be to get the
+           expectation value of the error operator with the Hartree-Fock
+           state or CISD state, which can scalably bound the error in
+           the ground state but much more accurately than the triangle
+           inequality.
+    """
+    if tight:
+        # Return the 1-norm of the error operator (upper bound on error).
+        return numpy.sum(numpy.absolute(
+            list(jellium_error_operator(
+                terms, indices, is_hopping_operator).terms.values())))
+
+    zero = FermionOperator()
+    error = 0.0
+
+    for alpha in range(len(terms)):
+        term_a = terms[alpha]
+        coefficient_a, = term_a.terms.values()
+        if coefficient_a:
+            error_a = 0.0
+
+            for beta in range(alpha + 1, len(terms)):
+                term_b = terms[beta]
+                coefficient_b, = term_b.terms.values()
+                # If the terms don't commute, add the coefficient.
+                if not commutator(term_a, term_b).isclose(zero):
+                    error_a += abs(coefficient_b)
+
+            error += 4.0 * abs(coefficient_a) * error_a ** 2
+
+    return error
+
+
+def ordered_jellium_terms_with_info(
+        grid_length, dimension=3, wigner_seitz_radius=10.,
+        inverse_filling_fraction=2, spinless=True):
+    """Give terms from the jellium Hamiltonian in simulation algorithm order.
+
+    Roughly uses the simulation ordering. Pre-computes term info.
```

Comment:
Please expand explanation to make it more clear that this is a preprocessing step for the error operator calculation.